contracts/Bonding.sol

```diff
@@ -97,85 +106,138 @@ contract Bonding is CollectableDust {
-            // The deposit must be a multiple of the difference between the stop
-            // time and the start time
-            uint256 streamDuration = streamStop - streamStart;
-            tokenAmount = (tokenAmount / streamDuration) * streamDuration;
-
-            IERC20(manager.uADTokenAddress()).safeApprove(address(sablier), 0);
-            IERC20(manager.uADTokenAddress()).safeApprove(
-                address(sablier),
-                tokenAmount
-            );
-
-            sablier.createStream(
-                msg.sender,
-                tokenAmount,
-                manager.uADTokenAddress(),
-                streamStart,
-                streamStop
-            );
-        }
+        // console.log("_sharesAmount", _sharesAmount);
+        // console.log("tokenAmount", tokenAmount);
+        // if (redeemStreamTime == 0) {
+        IERC20(manager.stableSwapMetaPoolAddress()).safeTransfer(
+            msg.sender,
+            _sharesAmount.redeemBonds(_currentShareValue, TARGET_PRICE)
+        );
+        //     } else {
+        //         // The transaction must be processed by the Ethereum blockchain before
+        //         // the start time of the stream, or otherwise the sablier contract
+        //         // reverts with a "start time before block.timestamp" message.
+        //         uint256 streamStart = block.timestamp + 60; // tx mining + 60 seconds
+        //         uint256 streamStop = streamStart + redeemStreamTime;
+        //         // The deposit must be a multiple of the difference between the stop
+        //         // time and the start time
+
+        //         uint256 streamDuration = streamStop - streamStart;
+        //         tokenAmount = (tokenAmount / streamDuration) * streamDuration;
+
+        //         IERC20(manager.stableSwapMetaPoolAddress()).safeApprove(
+        //             address(sablier),
+        //             0
+        //         );
+        //         IERC20(manager.stableSwapMetaPoolAddress()).safeApprove(
+        //             address(sablier),
+        //             tokenAmount
+        //         );
+        //         sablier.createStream(
+        //             msg.sender,
+        //             tokenAmount,
+        //             manager.stableSwapMetaPoolAddress(),
+        //             streamStart,
+        //             streamStop
+        //         );
+        //     }
     }
 
+    // solhint-disable-next-line no-empty-blocks
     function redeemAllShares() public {
-        redeemShares(
-            IERC20(manager.bondingShareAddress()).balanceOf(msg.sender)
-        );
+        // for each id of msg.sender
+        // redeemShares(
+        //     IERC20(manager.bondingShareAddress()).balanceOf(msg.sender),
+        //     id
+        // );
     }
 
-    function currentShareValue() public view returns (uint256 pricePerShare) {
+    // SI totalShares = 0  priceShare = TARGET_PRICE
```

Comment:
translate to EN

Reply:
can you provide a more literal description for the formulas and use Doxygen-style tags e.g
 /// @dev
  /// @param
   /// @returns
    /// @notice